#!/usr/bin/env python
PACKAGE = 'cpt_planning_ros'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("max_a", double_t, 0, "max acceleration", 0.1, 0.05, 3)
gen.add("max_v", double_t, 0, "max velocity", 1, 0.5, 3)
gen.add("output_enable", bool_t, 0, "Enable trajectory output", False)
gen.add("joystick_xy_scaling", double_t, 0, "Joystick Scaling", 0.1, 0.05, 1)
gen.add("joystick_z_scaling", double_t, 0, "Joystick Scaling", 0.1, 0.05, 1)
gen.add("terrain_min_dist", double_t, 0, "Minimum distance to terrain", 0.5, 0.01, 2.0)
gen.add("terrain_max_dist", double_t, 0, "Maximum distance to terrain", 0.3, 3.0, 5.0)
gen.add("updateOdomFromCurrentRef", bool_t, 0, "Update position from reference", True)
gen.add("updateOdomVel", bool_t, 0, "Update velocity from odometry", True)
gen.add("terrain_mode", bool_t, 0, "Fly relative to terrain", True)

gen.add("terrain_alpha_normal", double_t, 0, "Alpha normal", 6.0, 0, 20)
gen.add("terrain_beta_normal", double_t, 0, "Beta normal", 8.0, 0, 20)
gen.add("terrain_gamma_normal", double_t, 0, "Gamma normal", 0.14, 0, 2)

gen.add("terrain_alpha_along", double_t, 0, "Alpha along", 0.6, 0, 10)
gen.add("terrain_beta_along", double_t, 0, "Beta along", 8.0, 0, 20)
gen.add("terrain_gamma_along", double_t, 0, "Gamma along", 0.56, 0, 2)

gen.add("freeflight_alpha", double_t, 0, "N/A", 0.6, 0, 10)
gen.add("freeflight_beta", double_t, 0, "N/A", 0.6, 0, 10)
gen.add("freeflight_gamma", double_t, 0, "N/A", 0.6, 0, 10)

gen.add("obstacle_enable", double_t, 0, "N/A", 0.6, 0, 10)
gen.add("obstacle_nu_rep", double_t, 0, "N/A", 0.6, 0, 10)
gen.add("obstacle_nu_damp", double_t, 0, "N/A", 0.6, 0, 10)
gen.add("obstacle_eta_rep", double_t, 0, "N/A", 0.6, 0, 10)
gen.add("obstacle_eta_damp", double_t, 0, "N/A", 0.6, 0, 10)
gen.add("obstacle_gamma_damp", double_t, 0, "N/A", 0.6, 0, 10)
gen.add("obstacle_r", double_t, 0, "N/A", 0.6, 0, 10)

exit(gen.generate(PACKAGE, "cpt_planning_ros", "RMPConfig"))
